### PR TITLE
Use commons unit_names for ships

### PR DESCRIPTION
**Pull request by cetvrtak**

### CODE REVIEW

File: EU4toV2/Source/V2World/Army/Army.cpp [hunk 1]
```diff
@@ -281,6 +283,10 @@ V2::AddRegimentToArmyResult V2::Army::addRegimentToArmy(const REGIMENTTYPE chose
 		// Assign a national name ("1st Bavarian Frigate")
 		regiment.setName(getRegimentName(chosenType, unitNameCount, localAdjective));
 	}
+	if (unitNames)
+	{
+		regiment.nameShip(unitNames);
+	}
```
Comment (cetvrtak) [inline EU4toV2/Source/V2World/Army/Army.cpp:289]: @Zemurin This is all the change. Everything else is just importing/outputting unit names and helper functions for setting the ship name.

File: EU4toV2/Source/V2World/Country/CountryDetails.cpp [hunk 2]
```diff
@@ -54,8 +54,8 @@ void V2::CountryDetails::registerKeys()
 		parties.push_back(newParty);
 	});
 	registerKeyword("unit_names", [this](const std::string& unused, std::istream& theStream) {
-		const commonItems::stringOfItem unitNamesStr(theStream);
-		unitNames = unitNamesStr.getString();
+		const UnitNames newUnitNames(theStream);
```
Comment (Zemurin) [inline EU4toV2/Source/V2World/Country/CountryDetails.cpp:57]: unitNames = std::make_shared<UnitNames>(theStream);

File: EU4toV2/Source/V2World/Country/ModCommons.cpp [hunk 3]
```diff
@@ -18,7 +18,8 @@ V2::ModCommons::ModCommons(const std::string& tag, const std::string& filename)
 		parties.push_back(newParty);
 	});
 	registerKeyword("unit_names", [this](const std::string& unused, std::istream& theStream) {
-		unitNames = commonItems::stringOfItem(theStream).getString();
+		const UnitNames newUnitNames(theStream);
```
Comment (Zemurin) [inline EU4toV2/Source/V2World/Country/ModCommons.cpp:21]: ditto

File: EU4toV2/Source/V2World/Country/UnitNames.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
+#include "UnitNames.h"
+#include "CommonRegexes.h"
+#include "ParserHelpers.h"
+
+V2::UnitNames::UnitNames(std::istream& theStream)
+{
+	registerKeyword("dreadnought", [this](const std::string& unused, std::istream& theStream) {
+		dreadnoughts = commonItems::stringList(theStream).getStrings();
```
Comment (Zemurin) [inline EU4toV2/Source/V2World/Country/UnitNames.cpp:8]: dreadnoughts = commonItems::getStrings(theStream);

Ditto below.

**Review by Zemurin — CHANGES_REQUESTED** [review]
just a few stylistic changes

**Pull request merged**